src/runtime/particle-execution-host.ts

```diff
@@ -47,262 +47,301 @@ export class ParticleExecutionHost {
   private idlePromise: Promise<Map<Particle, number[]>> | undefined;
   private idleResolve: ((relevance: Map<Particle, number[]>) => void) | undefined;
 
-  constructor(port, slotComposer: SlotComposer, arc: Arc) {
+  constructor(port, slotComposer: SlotComposer, arc: Arc, javaPort: Port) {
```

Comment:
rather than requiring 2 ports, you should set this up to allow a variadic number of ports..

Reply:
i used an array instead - it makes it easier to filter out the javaport, if it is not provided (unless i'm ignorantly missing out on some elegant syntax)